pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -80,6 +80,10 @@ func (r *Registration) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeCla
 
 func (r *Registration) syncNode(ctx context.Context, nodeClaim *v1beta1.NodeClaim, node *v1.Node) error {
 	stored := node.DeepCopy()
+	// Strip the resource version off the original node. This ensures the patch will fail with a conflict if the node
+	// had been updated between the original GET and the PATCH. This is necessary because .spec.taints does not specify
+	// a patchMergeKey and patchStrategy. Ref: https://github.com/kubernetes/kubernetes/pull/113136
+	stored.ResourceVersion = ""
```

Comment:
Can we consider just doing an Update here rather than leveraging this "hack" on the PATCH API in upstream? Though I think it is a cool thing to use, I question whether it makes more sense for us to just use the UPDATE API since this behavior is well-known, including it's well-known how conflicts get handled with these calls

Reply:
What's our strategy for requeue limits? Until it hits the registration TTL and is then deleted?